src/Entities/Project.php

```diff
@@ -37,6 +57,27 @@ public function getIsOverdue(): ?bool
         return $this->isOverdue;
     }
 
+    public function toAssociativeArrayAllProperties(): array
+    {   
+        $usersArray = $this->users;
+
+        foreach($usersArray as $user){
+            //this method comes from User class
+            $users[] = $user->toAssociativeArray();
+        }
+
+        return [
+            'id' => strval($this->id),
+            'name' => $this->name, 
+            'client_id' => strval($this->client_id),
+            'deadline' => is_null($this->deadline) ? null : $this->deadline->format('d/m/Y'), 
+            'overdue' => $this->isOverdue,
+            'client_name' => $this->client_name,
+            'client_logo' => (empty($this->client_logo)) ? '' : $this->client_logo,
```

Comment:
we are wondering whether this is still necessary now? given the error before may have been because of the faulty code in the constructor

Reply:
From further looking, we think it will be fine to set it as $this->client_logo [https://code.tutsplus.com/tutorials/php-isset-vs-empty-vs-is_null--cms-37162](url)